src/net.md

```diff
@@ -1019,6 +1020,93 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-progress-with-range]: #ex-progress-with-range
+<a name="ex-progress-with-range"/>
+## Make a partial download with HTTP range headers
+
+[![reqwest-badge]][reqwest] [![cat-net-badge]][cat-net]
+
+Uses [`reqwest::Client::head`] to get the content-length and validate if the server sets the header
+[`reqwest::header::ContentRange`], required to confirm the support of partial downloads.
+
+If supported downloads the content using [`reqwest::get`], setting the [`reqwest::header::Range`]
+to do partial downloads in chunks of 100 bytes, between these writes basic progress messages.
+
+Range header, defined in [RFC7233][HTTP Range RFC7233].
+
+```rust,no_run
+# #[macro_use]
+# extern crate error_chain;
+extern crate reqwest;
+
+use std::io::Read;
+
+use reqwest::header::{Range, ContentRange, ContentRangeSpec};
+
+# error_chain! {
+#     foreign_links {
+#         Io(std::io::Error);
+#         Reqwest(reqwest::Error);
+#     }
+# }
+
+fn run() -> Result<()> {
+    let client = reqwest::Client::new();
+
+    let url = "https://httpbin.org/range/1024?duration=2";
+    let response = client.head(url).send()?;
+
+    let range = response.headers().get::<ContentRange>().ok_or(
+        "response doesn't include the expected ranges",
+    )?;
+
+    let content_length = match **range {
+        ContentRangeSpec::Bytes { instance_length: Some(len), .. } => len,
+        _ => bail!("response doesn't include the expected ranges"),
+    };
+
+    let mut content: Vec<u8> = Vec::with_capacity(content_length as usize);
```

Comment:
most of the time when we use partial downloading it's due to the fact that we either want to resume the download later or the whole content is to larg to hold in a contiguous memory buffer. 

I would suggest using `content` buffer length as the each partial content request length (upto the last possibly shorter one). and each time writing it into a file with given offset.

Reply:
I'll use the value of the constant to set the size of the buffer. 

What will be the filename? wouldn't that be to much for this example, I'm just concerned about the amount of code for the reqwest/ranges compared to the file handle.

Reply:
opening and writing to a file would be only two additional lines which we could hide with `#` anyway. the filename can be anything like "download.bin"